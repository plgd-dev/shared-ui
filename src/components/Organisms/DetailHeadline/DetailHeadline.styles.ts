import { css } from '@emotion/react'
import { HeadlineType } from './DetailHeadline.types'
import { ThemeType, getThemeColor, getTheme } from '../../Atomic/_theme'
import { fontPrimary } from '../../Atomic/_utils/commonStyles'

const getSizes = (type: HeadlineType) => {
    switch (type) {
        case 'h4':
            return css`
                font-size: 14px;
                line-height: 16px;
            `
    }
}

export const headline = (theme: ThemeType, type: HeadlineType) => css`
    color: ${getThemeColor(theme, `DetailHeadline.color`)};
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    font-style: normal;
    font-weight: 600;
    ${getSizes(type)};
    margin: 32px 0 16px 0;
`
