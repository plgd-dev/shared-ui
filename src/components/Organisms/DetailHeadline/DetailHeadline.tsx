import { createElement, FC } from 'react'
import classNames from 'classnames'

import { Props, defaultProps } from './DetailHeadline.types'
import * as styles from './DetailHeadline.styles'

const DetailHeadline: FC<Props> = (props) => {
    const mergedProps = { ...defaultProps, ...props }
    const HeadlineElement = ({ type, children, ...props }: Props) => createElement(type as string, props, children)

    const { dataTestId, className, ...rest } = mergedProps

    return (
        <HeadlineElement
            {...rest}
            children={mergedProps.children}
            className={classNames(mergedProps.type, className)}
            css={[(theme) => styles.headline(theme, mergedProps.type!)]}
            data-test-id={mergedProps.dataTestId}
        />
    )
}

DetailHeadline.displayName = 'DetailHeadline'

export default DetailHeadline
