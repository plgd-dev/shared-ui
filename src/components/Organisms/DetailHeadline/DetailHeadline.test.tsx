import { render } from '@testing-library/react'
import DetailHeadline from './DetailHeadline'

describe('DetailHeadline component', () => {
    it('renders without crashing', () => {
        const { container, asFragment } = render(<DetailHeadline type='h1'>Headline</DetailHeadline>)
        expect(container).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders correct headline type', () => {
        const { getByRole } = render(<DetailHeadline type='h1'>Test Headline</DetailHeadline>)
        expect(getByRole('heading', { level: 1 })).toBeInTheDocument()
    })

    it('applies custom class name', () => {
        const { container } = render(
            <DetailHeadline className='custom-class' type='h1'>
                Test Headline
            </DetailHeadline>
        )
        expect(container.firstChild).toHaveClass('custom-class')
    })

    it('renders children correctly', () => {
        const { getByText } = render(<DetailHeadline type='h1'>Test Headline</DetailHeadline>)
        expect(getByText('Test Headline')).toBeInTheDocument()
    })

    it('applies data-test-id correctly', () => {
        const { getByTestId } = render(
            <DetailHeadline dataTestId='test-id' type='h1'>
                Test Headline
            </DetailHeadline>
        )
        // container.querySelectorAll('[data-test-id="button-test"]')
        expect(getByTestId('test-id')).toBeInTheDocument()
    })
})
