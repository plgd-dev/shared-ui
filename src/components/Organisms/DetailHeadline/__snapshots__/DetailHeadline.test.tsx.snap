// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailHeadline component renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  color: #0A2965;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  margin: 32px 0 16px 0;
}

<h1
    class="h1 emotion-0"
  >
    Headline
  </h1>
</DocumentFragment>
`;
